py-glaredb/src/session.rs

```diff
@@ -22,7 +18,6 @@ use crate::{error::PyGlareDbError, logical_plan::PyLogicalPlan, runtime::wait_fo
 #[pyclass]
 pub struct LocalSession {
     pub(super) sess: PyTrackedSession,
-    pub(super) engine: Engine,
```

Comment:
I believe we'll want to keep this around so that's not accidentally dropped. The engine contains the background jobs worker and the metastore client workers.

Reply:
Makes sense; I brought back the engine and the `close` method, though now it only does the engine shutdown.